hs_core/hydroshare/utils.py

```diff
@@ -493,17 +514,23 @@ def copy_resource_files_and_AVUs(src_res_id, dest_res_id, set_to_private=False):
                 new_file_path = ori_file_path[0:idx1] + \
                                 settings.HS_LOCAL_PROXY_USER_IN_FED_ZONE + \
                                 '/' + tgt_res.short_id + ori_file_path[idx2:]
-                ResourceFile.objects.create(content_object=tgt_res,
-                                            fed_resource_file=new_file_path)
+                new_resource_file = ResourceFile.objects.create(content_object=tgt_res,
+                                                                fed_resource_file=new_file_path)
         elif f.resource_file:
             ori_file_path = f.resource_file.name
             new_file_path = tgt_res.short_id + ori_file_path[res_id_len:]
-            ResourceFile.objects.create(content_object=tgt_res, resource_file=new_file_path)
+            new_resource_file = ResourceFile.objects.create(content_object=tgt_res,
+                                                            resource_file=new_file_path)
+        # if the original file is part of a logical file, then
+        # add the corresponding new resource file to the copy of that logical file
+        if f.has_logical_file:
+            trg_logical_file = map_logical_files[f.logical_file]
+            trg_logical_file.add_resource_file(new_resource_file)
```

Comment:
@pkdash Can you rename the variable to ```tgt_logical_file``` to be variable name consistent within this method?